lib/cosmos/tools/script_runner/script_runner_frame.rb

```diff
@@ -962,6 +965,10 @@ def show_debug
               end
 
               if @script_binding
+                # Check for accessing an instance variable or local
+                if debug_text =~ /^@\S+$/ || @script_binding.local_variables.include?(debug_text.to_sym)
+                  debug_text = "puts #{debug_text}" # Automatically add puts to print it
+                end
```

Comment:
We also need to see if we can make setting local variables work from the debug prompt.  That didn't work in at least one case during the training we did.

Reply:
Seems to work in a little test I ran

Reply:
It doesn't work if you set a local variable in the debugger, and then let the script run to the point where it is used (With no other declaration).  Not sure if we can make that case work or not

Reply:
You can set a local on the binding but that doesn't help for the generic case because you can't push back locals into Object. See this [StackOverflow](https://stackoverflow.com/questions/18552891/how-to-dynamically-create-a-local-variable) post.

For example I tried this:
```
var = debug_text.split('=')[0].strip
@script_binding.local_variable_set(var.to_sym, 0)
```
And while that works it doesn't help because as soon as you hit the undeclared variable it complains:
```
ERROR: NameError : undefined local variable or method `x' for Object:Class
```